src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -1000,12 +1000,43 @@ void NotationViewInputController::mouseMoveEvent(QMouseEvent* event)
         }
     }
 
-    // move canvas
     if (!isNoteEnterMode || isMiddleButton) {
+        // move canvas
         PointF logicalDragDelta = logicPos - m_mouseDownInfo.logicalBeginPoint;
         m_view->moveCanvas(logicalDragDelta.x(), logicalDragDelta.y());
-
         m_isCanvasDragged = true;
+
+        // wrap mouse over the edge of the canvas
+        QPoint newPos = event->pos();
+
+        int right = m_view->width();
+        int bottom = m_view->height();
+
+        int padding = 2; // This is the minimum magic number to get the cursor to wrap over the right edge when muse score is fullscreen on X11
+        if (newPos.x() < padding) {
+            newPos.setX(right - padding);
+        } else if (newPos.x() + padding > right) {
+            newPos.setX(padding);
+        }
+        if (newPos.y() < padding) {
+            newPos.setY(bottom - padding);
+        } else if (newPos.y() + padding > bottom) {
+            newPos.setY(padding);
+        }
+
+        QPoint delta = newPos - event->pos();
+        if (delta.x() != 0 || delta.y() != 0) {
+            QPoint newCursorPos = cursor.pos() + delta;
+            cursor.setPos(newCursorPos);
+            QPoint cursorDelta = cursor.pos() - newCursorPos;
```

Comment:
`pos` and `setPos` are static methods of QCursor, so you don't need to (and shouldn't) create a QCursor instance, but just write `QCursor::pos()` and `QCursor::setPos(…)`

Reply:
This should be fixed now thanks